packages/linear-progress/src/mwc-linear-progress.ts

```diff
@@ -0,0 +1,125 @@
+/**
+@license
+Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {BaseElement, html, property, observer, query, customElement, Adapter, Foundation} from '@material/mwc-base/base-element';
+import {style} from './mwc-linear-progress-css.js';
+import MDCLinearProgressFoundation from '@material/linear-progress/foundation';
+
+export interface LinearProgressFoundation extends Foundation {
+  setDeterminate(value: boolean): void;
+  setProgress(value: number): void;
+  setBuffer(value: number): void;
+  setReverse(value: boolean): void;
+  open(): void;
+  close(): void;
+}
+
+export declare var LinearProgressFoundation: {
+  prototype: LinearProgressFoundation;
+  new(adapter: Adapter): LinearProgressFoundation;
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'mwc-linear-progress': LinearProgress;
+  }
+}
+
+@customElement('mwc-linear-progress' as any)
+export class LinearProgress extends BaseElement {
+  protected mdcFoundation!: LinearProgressFoundation;
+
+  protected readonly mdcFoundationClass: typeof LinearProgressFoundation = MDCLinearProgressFoundation;
+
+  @query('.mdc-linear-progress')
+  mdcRoot!: HTMLElement
+
+  @query(MDCLinearProgressFoundation.strings.PRIMARY_BAR_SELECTOR)
```

Comment:
Pull this and `BUFFER_SELECTOR` up to the top with object spread:

`const {PRIMARY_BAR_SELECTOR, BUFFER_SELECTOR} = MDCLinearProgressFoundation.strings`